### ISSUE TITLE
Add support for VisualStudio 2017

### ISSUE THREAD

**Issue by ghale** [gradle/gradle-native#123]
As a user, I want to be able to build and execute components built with the Visual Studio 2017 tools and runtime libraries.

## Implementation notes
- See discussion/decision on auto-detection issues at #34 

## Not Covered
- Any additional modeling of runtime libraries or sdks (i.e. this is only to get VisualStudio 2017 working with the current modeling of toolchains)

## Test Cases
- All of the standard language-specific tests for building, installing, and executing native components work with VS2017.
- Toolchain discovery can find an installed version of VS2017.
- User can specify the location of a VS2017 installation.

**Comment by ghale** [comment]
This functionality is not complete, but I wanted to get some changes pushed out and visible:

https://github.com/gradle/gradle/tree/gh/stable-native/vs2017-support

Basically, I've busted up `DefaultVisualStudioLocator` and decoupled the selection strategy from the installation/detection strategies which have been moved to `CommandLineVersionLocator` and `WindowsRegistryVersionLocator`.  At this point, it can detect VS2017 and earlier installations and can determine the compiler locations for each.  It can also take a user-specified install dir or a compiler on the path and derive the compatibility of the installation (basically either 2017-compatible or something-earlier-than-2017).  However, at the moment it can only actually use a version <= 2015.

The next step will be to construct a different `VisualStudioInstall` based on the compatibility of the installation and provide something back to the toolchain that accommodates the appropriate directory layout/tool locations for the given version.

A bit more to do here (and some more polish) before it is ready for review, but I'd appreciate any early feedback anyone might have on what I've done so far.
